### PR TITLE
docs: add "use markdown content in components" guide

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR adds a new guide called "Use markdown content in components".

The purpose is to address the question [I asked on Spectrum](https://spectrum.chat/mdx/general/react-component-markdown-as-a-prop~66bae3e4-bb27-4537-b3b3-ee3712a6eb49).

I also [wrote a blog post](https://joeprevite.com/writing-custom-note-component-mdx) about this (I wasn't sure if that should be included)

### CODE REVIEW

File: docs/guides/index.mdx [hunk 1]
```diff
@@ -10,6 +10,7 @@ to add to them by [contributing](/contributing)!
 - [Render MDX to the terminal](/guides/terminal)
 - [**\[WIP\]** Implement a dynamic table of contents](/guides/table-of-contents)
 - [**\[WIP\]** Use MDX in a Vue project](/guides/vue)
+- [Use markdown content in components ](/guides/markdown-in-components)
```
Comment (jsjoeio) [inline docs/guides/index.mdx:13]:
```suggestion
- [Use markdown content in components](/guides/markdown-in-components)
```

**Comment by wooorm** [comment]
Nice! ✨

Our linting is a bit strict; it’s warning about MDX style too. There are a couple things you can preview on the CI: https://github.com/mdx-js/mdx/pull/1060/checks?check_run_id=655230378#step:8:27. If you prefer local, `yarn lint` does the trick I believe!

**Comment by wooorm** [comment]
(there are currently some unrelated errors to your changes too btw, ignore those!)

**Comment by jsjoeio** [comment]
Ah got it! Okay I think I fixed mine. I tried to fix a few others (some worked, some didn't 😅)

**Comment by johno** [comment]
This is great, thanks a bunch  @jsjoeio!

**Pull request merged**